src/StudentOrganizer.Infrastructure/Services/TeamService.cs

```diff
@@ -1,52 +1,75 @@
-﻿using System.Linq;
-using System.Threading.Tasks;
-using StudentOrganizer.Core.Models;
+﻿using StudentOrganizer.Core.Models;
 using StudentOrganizer.Core.Repositories;
 using StudentOrganizer.Infrastructure.Commands.Teams;
 using StudentOrganizer.Infrastructure.IServices;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
 
 namespace StudentOrganizer.Infrastructure.Services
 {
-	public class TeamService : ITeamService
-	{
-		private readonly IGroupRepository _groupRepository;
-		private readonly IAdministratorService _administratorService;
+    public class TeamService : ITeamService
+    {
+        private readonly IGroupRepository _groupRepository;
+        private readonly IAdministratorService _administratorService;
+        private readonly IUserRepository _userRepository;
+      
+
+        public TeamService(IGroupRepository groupRepository, IAdministratorService administratorService, IUserRepository userRepository)
+        {
+            _groupRepository = groupRepository;
+            _administratorService = administratorService;
+            _userRepository = userRepository;
+        }
+
+        public async Task AddUsersToTeam(AddUsersToTeam command)
+        {
+            await _administratorService.ValidateAdministrativePrivileges(command.UserId, command.GroupId);
+            var group = await _groupRepository.GetWithTeamsAndStudentsAsync(command.GroupId);
+            var users = await _userRepository.GetUsersByEmails(command.Emails);
```

Comment:
Zamiast pobierać tutaj userów mogłeś wziąc z groupRepository bo tak to dwa query idą do bazy danych